plugin.video.cbc-sports/addon.py

```diff
@@ -0,0 +1,417 @@
+	sdata = str(get_html(stream))
+
+	try:
+	    errfound = sdata.find('An error occurred')
+	except IndexError:
+            xbmcgui.Dialog().notification(name, translation(30010), defaultimage, notetime, False)
+            return
+
+	if errfound > -1:
+		xbmcgui.Dialog().notification(name, translation(30010), defaultimage, notetime, False)
+		return
+	if cbclog >= 1:
+	    xbmc.log('CBC Sports Live Schedule stream return data: ' + sdata, xbmc.LOGINFO)
+
+	try:	
+	    xbmc.Player().play( stream, listitem )
+	except:              
+	    xbmcgui.Dialog().notification(name, translation(30010), defaultimage, notetime, False)
+
+	sys.exit()
+	xbmcplugin.endOfDirectory(int(sys.argv[1]))
+
+
+#6
+def VIDEOS(url):
+	jresponse = urllib.request.urlopen(url)
+	cbclog = int(xbmcaddon.Addon().getSetting('cbclog'))
+	jdata = json.load(jresponse);i=0
+	if cbclog >= 1:
+	    xbmc.log('CBC Sports VIDEOS name: ' + str(url), xbmc.LOGINFO)
+	if cbclog == 2:
+	    xbmc.log('CBC Sports VIDEOS data ' + str(jdata), xbmc.LOGINFO)
+	item_dict = jdata
+	count = len(item_dict['entries'])
+	for item in jdata['entries']:
+		title = (jdata['entries'][i]['title'])
+		url = jdata['entries'][i]['content'][0]['url']
+		image = jdata['entries'][i]['defaultThumbnailUrl']
+		vduration = int(jdata['entries'][i]['content'][0]['duration'])
+		pubDate = jdata['entries'][i]['pubDate']
+		aired = datetime.fromtimestamp(pubDate / 1000).strftime('%Y-%m-%d')
+		plot = jdata['entries'][i]['description']
+		vheight = int(jdata['entries'][i]['content'][0]['height'])
+		vwidth = int(jdata['entries'][i]['content'][0]['width'])
+		#xbmc.log('CBC Sports Live Schedule stream aired ' + aired, xbmc.LOGINFO)
+		#xbmc.log('CBC Sports Live Schedule stream dplot ' + plot, xbmc.LOGINFO)
+		#xbmc.log('CBC Sports Video height and width: ' + str(vheight) + ' '  + str(vwidth), xbmc.LOGINFO)
+		addDir2(title, url, vduration, 7, image, aired, plot, vheight, vwidth);i=i+1
+	xbmcplugin.endOfDirectory(int(sys.argv[1]))
+		
+
+#7
+def GET_STREAM(name,url):
+	cbclog = int(xbmcaddon.Addon().getSetting('cbclog'))
+	if cbclog >= 1:
+	    xbmc.log('CBC Sports GET_STREAM name: ' + str(name), xbmc.LOGINFO)
+	    xbmc.log('CBC Sports URL: ' + str(url), xbmc.LOGINFO)
+	smil = get_html(url)
+	contents = BeautifulSoup(smil,'html5lib')
+	if cbclog >= 1:
+	    xbmc.log('CBC Sports GET_STREAM contents: ' + str(contents), xbmc.LOGINFO)
+	if 'GeoLocationBlocked' in str(contents):		#  Check for blackout
+	    xbmcgui.Dialog().notification(name, translation(30001), defaultimage, notetime, False)
+	    return
+	stream = (re.compile('src="(.+?)"').findall(str(contents))[0])
+	if mp4base not in stream:
+		#stream = mp4base + stream
+		stream = stream
+	listitem = xbmcgui.ListItem(name)
+	listitem.setArt({'thumb': defaultimage, 'icon': defaultimage})
+	if cbclog >= 1:
+	    xbmc.log('CBC Sports Live Schedule Playback stream: ' + str(stream), xbmc.LOGINFO)
+	xbmc.Player().play( stream, listitem )
```

Comment:
Same

Reply:
Code has been updated to use xbmcplugin.setResolvedUrl